libs/auth/src/artifact_authentication.ts

```diff
@@ -247,10 +232,31 @@ function constructSignatureFilePath(artifact: ArtifactToImport): string {
 }
 
 async function performArtifactSpecificAuthenticationChecks(
-  artifact: ArtifactToImport
+  artifact: ArtifactToImport,
+  signingMachineDetails: MachineDetails
 ): Promise<void> {
   switch (artifact.type) {
     case 'cast_vote_records': {
+      assert(
+        signingMachineDetails.machineType === 'central-scan' ||
+          signingMachineDetails.machineType === 'scan',
+        'Signing machine for cast vote records should be a VxCentralScan or VxScan'
+      );
+
+      const exportDirectoryName = path.basename(artifact.directoryPath);
+      const exportDirectoryNameComponents =
+        parseCastVoteRecordReportExportDirectoryName(exportDirectoryName);
+      assert(
+        exportDirectoryNameComponents !== undefined,
+        `Error parsing export directory name: ${exportDirectoryName}`
+      );
+      assert(
+        exportDirectoryNameComponents.machineId ===
+          signingMachineDetails.machineId,
+        `Machine ID in export directory name doesn't match machine ID in signing machine cert: ` +
+          `${exportDirectoryNameComponents.machineId} != ${signingMachineDetails.machineId}`
+      );
```

Comment:
I contemplated hopping a level deeper and verifying the various machine ID related fields in the CVR files themselves but chose not to for a few reasons:

- The CDF fields that house machine ID are designed to hold multiple machine IDs in the case of a consolidated export, so not just the exporting machine ID
- We pull from the export directory name, not the file contents, to surface machine ID in the VxAdmin import modal
- We already validate that the contents of the CVR files are unaltered

From a UI perspective, with this change in place, a user will still see an altered machine ID in the VxAdmin import modal. But on click to import, will be shown our existing "Error authenticating cast vote records" error message.

Reply:
consider an attacker who has access to the key for `machine-1`. They thus can only create a CVR directory that says `machine-1`. But can the CVRs falsely say they come from `machine-2`? What will the final report say in VxAdmin? Will it ever surface `machine-2`? Or maybe there's a later check?

Reply:
The attacker won't be able to display any machine ID other than `machine-1` in the CVR import modal. But yes, they could put an arbitrary machine ID inside the CVRs (per our assumption that they have access to machine 1's private key and can sign arbitrary data blobs).

I didn't want to over-rotate on this case because we eventually will have cases where a CVR contains machine IDs other than the signing machine, namely when VxAdmin will export aggregated post-adjudication CVRs.

But it seems like a fair assumption for now that VxScan will never export anything but its own CVRs. So I can add this check.

Reply:
Added valildation for the field used to determine and persist machine/scanner ID on import: https://github.com/votingworks/vxsuite/pull/5557/commits/857bfaf43179d48f6b9b25d264bbb28f5fdf9fc9